// styled-config/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "theme", {
  enumerable: true,
  get: function get() {
    return _theme.theme;
  }
});
Object.defineProperty(exports, "getBorder", {
  enumerable: true,
  get: function get() {
    return _utils.getBorder;
  }
});
Object.defineProperty(exports, "getBorderRadius", {
  enumerable: true,
  get: function get() {
    return _utils.getBorderRadius;
  }
});
Object.defineProperty(exports, "getBoxShadow", {
  enumerable: true,
  get: function get() {
    return _utils.getBoxShadow;
  }
});
Object.defineProperty(exports, "getColor", {
  enumerable: true,
  get: function get() {
    return _utils.getColor;
  }
});
Object.defineProperty(exports, "getFontFamily", {
  enumerable: true,
  get: function get() {
    return _utils.getFontFamily;
  }
});
Object.defineProperty(exports, "getFontSize", {
  enumerable: true,
  get: function get() {
    return _utils.getFontSize;
  }
});
Object.defineProperty(exports, "getFontWeight", {
  enumerable: true,
  get: function get() {
    return _utils.getFontWeight;
  }
});
Object.defineProperty(exports, "getHeight", {
  enumerable: true,
  get: function get() {
    return _utils.getHeight;
  }
});
Object.defineProperty(exports, "getMargin", {
  enumerable: true,
  get: function get() {
    return _utils.getMargin;
  }
});
Object.defineProperty(exports, "getPadding", {
  enumerable: true,
  get: function get() {
    return _utils.getPadding;
  }
});
Object.defineProperty(exports, "getConfigProperty", {
  enumerable: true,
  get: function get() {
    return _utils.getConfigProperty;
  }
});
Object.defineProperty(exports, "getBreakpointSize", {
  enumerable: true,
  get: function get() {
    return _utils.getBreakpointSize;
  }
});
Object.defineProperty(exports, "getConcreteBreakpointSize", {
  enumerable: true,
  get: function get() {
    return _utils.getConcreteBreakpointSize;
  }
});
Object.defineProperty(exports, "getWidth", {
  enumerable: true,
  get: function get() {
    return _utils.getWidth;
  }
});

var _theme = require("./theme");

var _utils = require("./utils");
