// styled-config/lib/theme/alert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.alert = void 0;
var alert = {
  colors: {
    primary: {
      text: function text(get) {
        return get('colorScheme', 'primaryDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'primaryLight');
      },
      border: function border(get) {
        return get('colorScheme', 'primaryLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'primaryLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'primaryDarkest');
      }
    },
    secondary: {
      text: function text(get) {
        return get('colorScheme', 'secondaryDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'secondaryLight');
      },
      border: function border(get) {
        return get('colorScheme', 'secondaryLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'secondaryLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'secondaryDarkest');
      }
    },
    success: {
      text: function text(get) {
        return get('colorScheme', 'successDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'successLight');
      },
      border: function border(get) {
        return get('colorScheme', 'successLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'successLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'successDarkest');
      }
    },
    danger: {
      text: function text(get) {
        return get('colorScheme', 'dangerDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'dangerLight');
      },
      border: function border(get) {
        return get('colorScheme', 'dangerLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'dangerLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'dangerDarkest');
      }
    },
    warning: {
      text: function text(get) {
        return get('colorScheme', 'warningDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'warningLight');
      },
      border: function border(get) {
        return get('colorScheme', 'warningLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'warningLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'warningDarkest');
      }
    },
    info: {
      text: function text(get) {
        return get('colorScheme', 'infoDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'infoLight');
      },
      border: function border(get) {
        return get('colorScheme', 'infoLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'infoLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'infoDarkest');
      }
    },
    light: {
      text: function text(get) {
        return get('colorScheme', 'lightDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'lightLight');
      },
      border: function border(get) {
        return get('colorScheme', 'lightDark');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'lightLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'darkDarkest');
      }
    },
    dark: {
      text: function text(get) {
        return get('colorScheme', 'darkDarkest');
      },
      background: function background(get) {
        return get('colorScheme', 'darkLight');
      },
      border: function border(get) {
        return get('colorScheme', 'darkLighter');
      },
      borderTop: function borderTop(get) {
        return get('colorScheme', 'darkLighter');
      },
      hover: function hover(get) {
        return get('colorScheme', 'darkDarkest');
      }
    },
    default: function _default(get) {
      return get('alert', 'colors', 'secondary');
    }
  },
  margin: {
    bottom: '1rem'
  },
  padding: {
    default: '0.75rem 1.25rem',
    dismissibleRight: '4rem'
  },
  border: {
    default: '1px solid transparent'
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  },
  fontWeight: {
    default: '700'
  }
};
exports.alert = alert;
