// styled-config/lib/theme/badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badge = void 0;
var badge = {
  colors: {
    primary: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'primary');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'primaryDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    secondary: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'secondary');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'secondaryDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    success: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'success');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'successDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    danger: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'danger');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'dangerDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    warning: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'warning');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'warningDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'darker');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'darker');
      }
    },
    info: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'info');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'infoDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    light: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'light');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'lightDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'darker');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'darker');
      }
    },
    dark: {
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'dark');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'darkDarker');
      },
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      linkColor: function linkColor(get) {
        return get('colorScheme', 'white');
      }
    },
    default: function _default(get) {
      return get('badge', 'colors', 'secondary');
    }
  },
  padding: {
    pill: '0.25rem 0.6rem',
    default: '0.25rem 0.4rem'
  },
  fontWeight: {
    default: '700'
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  }
};
exports.badge = badge;
