// styled-config/lib/theme/button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.button = void 0;
var button = {
  colors: {
    primary: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'primary');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'primary');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'primary');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'primaryDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'primaryDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'primary');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'primary');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'primaryDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'primaryDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'primaryBoxShadow');
      }
    },
    secondary: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'secondary');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'secondary');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'secondary');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'secondaryDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'secondary');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'secondary');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'secondary');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'secondaryDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'secondaryDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'secondaryBoxShadow');
      }
    },
    success: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'success');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'success');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'success');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'successDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'successDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'success');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'success');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'successDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'successDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'successBoxShadow');
      }
    },
    danger: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'danger');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'danger');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'danger');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'dangerDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'dangerDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'danger');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'danger');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'dangerDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'dangerDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'dangerBoxShadow');
      }
    },
    warning: {
      color: function color(get) {
        return get('colorScheme', 'darkerDarker');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'warning');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'darkerDarker');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'warning');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'warning');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'warningDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'warningDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'warning');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'warning');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'warningDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'warningDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'warningBoxShadow');
      }
    },
    info: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'info');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'info');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'info');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'infoDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'infoDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'info');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'info');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'infoDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'infoDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'infoBoxShadow');
      }
    },
    light: {
      color: function color(get) {
        return get('colorScheme', 'darkerDarker');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'light');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'darkerDarker');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'light');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'light');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'lightDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'light');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'light');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'lightDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'lightDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'lightBoxShadow');
      },
      toggle: {
        color: function color(get) {
          return get('colorScheme', 'dark50Alpha');
        },
        borderColor: function borderColor(get) {
          return get('colorScheme', 'dark10Alpha');
        },
        colorHoverFous: function colorHoverFous(get) {
          return get('colorScheme', 'dark50Alpha');
        },
        borderColorHoverFous: function borderColorHoverFous(get) {
          return get('colorScheme', 'dark10Alpha');
        }
      }
    },
    dark: {
      color: function color(get) {
        return get('colorScheme', 'white');
      },
      colorOutline: function colorOutline(get) {
        return get('colorScheme', 'dark');
      },
      colorOutlineHover: function colorOutlineHover(get) {
        return get('colorScheme', 'white');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'dark');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'dark');
      },
      backgroundColorHoverFocus: function backgroundColorHoverFocus(get) {
        return get('colorScheme', 'darkDarker');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'darkDark');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'dark');
      },
      borderColorDisabled: function borderColorDisabled(get) {
        return get('colorScheme', 'dark');
      },
      borderColorHoverFocus: function borderColorHoverFocus(get) {
        return get('colorScheme', 'darkDark');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'darkDark');
      },
      boxShadow: function boxShadow(get) {
        return get('colorScheme', 'darkBoxShadow');
      },
      toggle: {
        color: function color(get) {
          return get('colorScheme', 'white50Alpha');
        },
        borderColor: function borderColor(get) {
          return get('colorScheme', 'white10Alpha');
        },
        colorHoverFous: function colorHoverFous(get) {
          return get('colorScheme', 'white50Alpha');
        },
        borderColorHoverFous: function borderColorHoverFous(get) {
          return get('colorScheme', 'white10Alpha');
        }
      }
    },
    default: function _default(get) {
      return get('button', 'colors', 'secondary');
    }
  },
  margin: {
    blockTop: '0.5rem',
    dropDownToggleLeft: '0255rem'
  },
  padding: {
    sm: '0.25rem 0.5rem',
    lg: '0.5rem 1rem',
    default: '0.375rem 0.75rem',
    splitSmRight: '0.375rem',
    splitSmLeft: '0.375rem',
    splitLgRight: '0.75rem',
    splitLgLeft: '0.75rem',
    splitDefaultRight: '0.5625rem',
    splitDefaultLeft: '0.5625rem'
  },
  fontSize: function fontSize(get) {
    return get('globals', 'fontSize');
  },
  border: function border(get) {
    return get('globals', 'border');
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  }
};
exports.button = button;
