// styled-config/lib/theme/formControl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.formControl = void 0;
var formControl = {
  colors: {
    default: {
      color: function color(get) {
        return get('colorScheme', 'dark');
      },
      background: function background(get) {
        return get('colorScheme', 'white');
      },
      border: function border(get) {
        return get('colorScheme', 'darkLighter');
      },
      borderFocus: function borderFocus(get) {
        return get('colorScheme', 'darkLight');
      },
      boxShadowFocus: function boxShadowFocus(get) {
        return get('colorScheme', 'darkerBoxShadow');
      },
      placeholder: function placeholder(get) {
        return get('colorScheme', 'secondary');
      },
      backgroundColorDisabledReadonly: function backgroundColorDisabledReadonly(get) {
        return get('colorScheme', 'whiteDark');
      },
      borderValid: function borderValid(get) {
        return get('colorScheme', 'success');
      },
      formControlBoxShadowValid: function formControlBoxShadowValid(get) {
        return get('colorScheme', 'successBoxShadow');
      },
      borderInvalid: function borderInvalid(get) {
        return get('colorScheme', 'danger');
      },
      formControlBoxShadowInvalid: function formControlBoxShadowInvalid(get) {
        return get('colorScheme', 'dangerBoxShadow');
      }
    }
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  },
  border: function border(get) {
    return get('globals', 'border');
  },
  boxShadow: {
    default: '0 0 0 0.2rem'
  },
  padding: {
    smRight: '0',
    smLeft: '0',
    lgRight: '0',
    lgLeft: '0',
    smPadding: '0.25rem 0.5rem',
    lgPadding: '0.5rem 1rem',
    defaultPadding: '0.375rem 0.75rem'
  },
  fontSize: {
    sm: '0.875rem',
    lg: '1.25rem',
    default: '1rem'
  },
  height: {
    select: 'calc(2.25rem + 2px)',
    selectLg: 'calc(2.875rem + 2px)',
    selectSm: 'calc(1.8125rem + 2px)'
  }
};
exports.formControl = formControl;
