// styled-config/lib/theme/listGroupItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.listGroupItem = void 0;
var listGroupItem = {
  colors: {
    primary: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'primaryDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'primaryLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'primaryLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'primaryDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'primaryDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'primaryDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    secondary: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'secondaryDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'secondaryLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'secondaryLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'secondaryDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'secondaryDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'secondaryDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    success: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'successDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'successLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'successLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'successDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'successDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'successDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    info: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'infoDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'infoLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'infoLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'infoDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'infoDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'infoDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    warning: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'warningDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'warningLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'warningLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'warningDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'warningDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'warningDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    danger: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'dangerDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'dangerLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'dangerLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'dangerDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'dangerDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'dangerDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    light: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'lightDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'lightDark');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'lightLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'lightDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'lightDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'lightDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    dark: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDarker');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'darkDarkest');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'darkLighter');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'darkLighter');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'darkDarkest');
      },
      backgroundColorHoverFocusActive: function backgroundColorHoverFocusActive(get) {
        return get('colorScheme', 'darkDarkest');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'darkDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      }
    },
    default: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'lightDark');
      },
      backgroundColor: function backgroundColor(get) {
        return get('colorScheme', 'white');
      },
      backgroundColorDisabled: function backgroundColorDisabled(get) {
        return get('colorScheme', 'white');
      },
      backgroundColorAction: function backgroundColorAction(get) {
        return get('colorScheme', 'light');
      },
      backgroundColorHoverFocusAction: function backgroundColorHoverFocusAction(get) {
        return get('colorScheme', 'light');
      },
      color: function color(get) {
        return get('colorScheme', 'lightDarkest');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white');
      },
      colorAction: function colorAction(get) {
        return get('colorScheme', 'lightDarkest');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'weakBorder');
      },
      borderColorActive: function borderColorActive(get) {
        return get('colorScheme', 'primary');
      },
      backgroundColorActive: function backgroundColorActive(get) {
        return get('colorScheme', 'primary');
      }
    }
  },
  padding: {
    default: '0.75rem 1.25rem'
  },
  border: function border(get) {
    return get('globals', 'border');
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  }
};
exports.listGroupItem = listGroupItem;
