// styled-config/lib/theme/navbarLink.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.navbarLink = void 0;
var navbarLink = {
  colors: {
    dark: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'white25Alpha');
      },
      colorDisabledHoverFocus: function colorDisabledHoverFocus(get) {
        return get('colorScheme', 'white25Alpha');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'white90Alpha');
      },
      colorActiveHoverFocus: function colorActiveHoverFocus(get) {
        return get('colorScheme', 'white90Alpha');
      },
      color: function color(get) {
        return get('colorScheme', 'white50Alpha');
      },
      colorHoverFocus: function colorHoverFocus(get) {
        return get('colorScheme', 'white75Alpha');
      }
    },
    light: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'dark25Alpha');
      },
      colorDisabledHoverFocus: function colorDisabledHoverFocus(get) {
        return get('colorScheme', 'dark25Alpha');
      },
      colorActive: function colorActive(get) {
        return get('colorScheme', 'dark90Alpha');
      },
      colorActiveHoverFocus: function colorActiveHoverFocus(get) {
        return get('colorScheme', 'dark90Alpha');
      },
      color: function color(get) {
        return get('colorScheme', 'dark50Alpha');
      },
      colorHoverFocus: function colorHoverFocus(get) {
        return get('colorScheme', 'dark75Alpha');
      }
    },
    default: function _default(get) {
      return get('navbarLink', 'colors', 'light');
    }
  },
  padding: {
    brandTop: '0.3125rem',
    brandBottom: '0.3125rem'
  },
  fontSize: function fontSize(get) {
    return get('globals', 'fontSize');
  }
};
exports.navbarLink = navbarLink;
