// styled-config/lib/theme/navLink.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.navLink = void 0;
var navLink = {
  colors: {
    light: {
      colorDisabled: function colorDisabled(get) {
        return get('colorScheme', 'darkLighter');
      },
      colorDisabledHoverFocus: function colorDisabledHoverFocus(get) {
        return get('colorScheme', 'darkLighter');
      },
      colorTabsActive: function colorTabsActive(get) {
        return get('colorScheme', 'dark');
      },
      colorTabsActiveHoverFocus: function colorTabsActiveHoverFocus(get) {
        return get('colorScheme', 'darkDark');
      },
      colorPillsActive: function colorPillsActive(get) {
        return get('colorScheme', 'white');
      },
      colorPillsActiveHoverFocus: function colorPillsActiveHoverFocus(get) {
        return get('colorScheme', 'white');
      },
      color: function color(get) {
        return get('colorScheme', 'primary');
      },
      colorHoverFocus: function colorHoverFocus(get) {
        return get('colorScheme', 'primaryDarker');
      },
      backgroundColorPillsActive: function backgroundColorPillsActive(get) {
        return get('colorScheme', 'primary');
      },
      borderColorTabsActive: function borderColorTabsActive(get) {
        return "".concat(get('colorScheme', 'darkLight'), " ").concat(get('colorScheme', 'darkLight'), " ").concat(get('colorScheme', 'white'));
      },
      borderColorTabsHoverFocus: function borderColorTabsHoverFocus(get) {
        return "".concat(get('colorScheme', 'darkLight'), " ").concat(get('colorScheme', 'darkLight'), " ").concat(get('colorScheme', 'darkLight'));
      },
      borderColorTabsDisabled: function borderColorTabsDisabled(get) {
        return get('colorScheme', 'transparent');
      },
      borderColorPillsActive: function borderColorPillsActive(get) {
        return get('colorScheme', 'primary');
      },
      borderColor: function borderColor(get) {
        return get('colorScheme', 'transparent');
      }
    },
    dark: function dark(get) {
      return get('navLink', 'colors', 'light');
    },
    default: function _default(get) {
      return get('navLink', 'colors', 'light');
    }
  },
  padding: {
    default: '0.5rem'
  },
  borderRadius: function borderRadius(get) {
    return get('globals', 'borderRadius');
  },
  border: function border(get) {
    return get('globals', 'border');
  }
};
exports.navLink = navLink;
